File: generator.py
import random


def generate(size, max_value=10000):
    values = [random.randint(1, max_value) for _ in range(size)]

    solvable = random.choice([True, False])

    if solvable:
        target = sum(random.sample(values, k=random.randint(1, size)))
    else:
        target = random.randint(1, size * max_value)

    filename = f"test/input-s{size}-{'solvable' if solvable else 'unknown'}.txt"

    with open(filename, "w") as f:
        f.write(f"{size}\n")
        f.write(f"{target}\n")
        f.write(" ".join(str(v) for v in values) + "\n")

    print(f"Generated {filename}: size={size}, target={target}")


if __name__ == "__main__":
    COUNT = 10

    for i in range(COUNT):
        SIZE = random.randint(10, 500)
        generate(SIZE)

File: src/generator.rs
use rand::Rng;
use std::fs;
use std::io::{BufWriter, Write};
use std::time::Instant;

pub fn generate(size: usize, min_value: usize, max_value: usize, solvable: Option<bool>) {
    let start = Instant::now();
    let mut rng = rand::thread_rng();

    let solvable = solvable.unwrap_or_else(|| rng.gen_bool(0.5));

    let gen_start = Instant::now();
    let values: Vec<usize> = (0..size)
        .map(|_| rng.gen_range(min_value..=max_value))
        .collect();
    println!("Values generated in {:.3?}", gen_start.elapsed());

    let target: usize = if solvable {
        let k = rng.gen_range(1..=size);
        let indices = rand::seq::index::sample(&mut rng, size, k);
        indices.iter().map(|i| values[i]).sum()
    } else {
        rng.gen_range(1..=(size * max_value))
    };

    let label = if solvable { "solvable" } else { "unknown" };
    let filename = format!("test/input-{}-{}.txt", size, label);

    fs::create_dir_all("test").expect("Failed to create test directory");
    let file = fs::File::create(&filename).expect("Failed to create output file");
    let mut writer = BufWriter::new(file);

    writeln!(writer, "{}", size).unwrap();
    writeln!(writer, "{}", target).unwrap();

    for (i, v) in values.iter().enumerate() {
        if i > 0 {
            write!(writer, " ").unwrap();
        }
        write!(writer, "{}", v).unwrap();
    }
    writeln!(writer).unwrap();

    let elapsed = start.elapsed();
    println!(
        "Generated {}: size={}, target={} ({:.3?})",
        filename, size, target, elapsed
    );
}
